unidist/core/backends/mpi/core/worker/request_store.py

```diff
@@ -247,48 +247,14 @@ def process_get_request(self, source_rank, data_id, is_blocking_op=False):
         -----
         Request is asynchronous, no wait for the data sending.
         """
-        object_store = ObjectStore.get_instance()
-        async_operations = AsyncOperations.get_instance()
-        if object_store.contains(data_id):
-            if source_rank == communication.MPIRank.ROOT or is_blocking_op:
-                # The controller or a requesting worker is blocked by the request
-                # which should be processed immediatly
-                operation_data = object_store.get(data_id)
-                # We use a blocking send here because the receiver is waiting for the result.
-                communication.send_complex_data(
-                    mpi_state.comm,
-                    operation_data,
-                    source_rank,
-                )
-            else:
-                operation_type = common.Operation.PUT_DATA
-                if object_store.is_already_serialized(data_id):
-                    operation_data = object_store.get_serialized_data(data_id)
-                    # Async send to avoid possible dead-lock between workers
-                    h_list, _ = communication.isend_complex_operation(
-                        mpi_state.comm,
-                        operation_type,
-                        operation_data,
-                        source_rank,
-                        is_serialized=True,
-                    )
-                    async_operations.extend(h_list)
-                else:
-                    operation_data = {
-                        "id": data_id,
-                        "data": object_store.get(data_id),
-                    }
-                    # Async send to avoid possible dead-lock between workers
-                    h_list, serialized_data = communication.isend_complex_operation(
-                        mpi_state.comm,
-                        operation_type,
-                        operation_data,
-                        source_rank,
-                        is_serialized=False,
-                    )
-                    async_operations.extend(h_list)
-                    object_store.cache_serialized_data(data_id, serialized_data)
-
+        local_store = LocalObjectStore.get_instance()
+        if local_store.contains(data_id):
```

Comment:
What if data is in shared buffer but not in local store?

Reply:
The Get command can only be received by the owner of the data. If the rank is the owner of the data, it is always contained in local storage.